api/src/main/java/org/apache/iceberg/expressions/Expressions.java

```diff
@@ -305,6 +337,21 @@ public static <T> NamedReference<T> ref(String name) {
     return new NamedReference<>(name);
   }
 
+  /**
+   * Constructs a resolved reference for a given column.
+   *
+   * <p>The following are equivalent: equals("a", 5) and if a fieldId is 1 and equals(ref("a", 1),
+   * 5).
+   *
+   * @param name a column name
+   * @param fieldId the field ID of the column
+   * @param <T> the Java type of this reference
+   * @return a named reference
+   */
+  static <T> IDReference<T> ref(String name, int fieldId) {
```

Comment:
should we adjust the order and make `fieldId` as the first arg? For IDReference, field name is actually ignored. So put the essential arg first.